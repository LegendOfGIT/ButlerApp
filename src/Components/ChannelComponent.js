import React, { Component } from 'react';
import ChannelItem from './ChannelItemComponent';

const Channel = () => {
    return (
        <div className="Channel">
            <h2>Test-Stream</h2>
            <ChannelItem />
            <ChannelItem />
            <ChannelItem />
        </div>
    );
};

export default Channel;
