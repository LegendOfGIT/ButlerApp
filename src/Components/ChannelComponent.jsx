import React, { Component } from 'react';
import { connect } from 'react-redux';
import ChannelItemContainer from '../Containers/ChannelItemContainer';
import { fetchChannelData } from '../Actions/ChannelActions';
import PropTypes from 'prop-types';

class ChannelComponent extends Component {
    componentDidMount() {
        this.props.fetchChannelData();
    }

    render() {
        if (this.props.isLoading)
        {
            return (
                <div className="alert alert-info">
                    Loading channel...
                    <div className="fa fa-spin fa-spinner">&nbsp;</div>
                </div>
            );
        }

        if (this.props.hasError)
        {
            return (
                <div className="alert alert-danger">error while loading channel</div>
            );
        }

        return (
            <div className="Channel">
                <h2>{this.props.ChannelId}</h2>
                {
                    this.props.ChannelItemIds.map((channelItemId) => (
                        <ChannelItemContainer id={channelItemId} />
                    ))
                }
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        ChannelId: state.channel.ChannelId,
        ChannelItemIds: state.channel.InformationItemIds,
        hasError: state.channel.hasError,
        isLoading: state.channel.isLoading
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchChannelData: () => { dispatch(fetchChannelData(''));  }
    }
};

ChannelComponent.propTypes = {
    fetchChannelData: PropTypes.func,
    ChannelId: PropTypes.string,
    ChannelItemIds: PropTypes.array,
    hasError: PropTypes.bool,
    isLoading: PropTypes.bool
};

ChannelComponent.defaultProps = {
    ChannelItemIds: []
};

export default connect(mapStateToProps, mapDispatchToProps)(ChannelComponent);
