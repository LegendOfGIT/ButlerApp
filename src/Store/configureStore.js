import { createStore, applyMiddleware } from 'redux';
import rootReducer from '../Reducers';
import thunk from 'redux-thunk';
import { compose } from 'redux';

export default function configureStore(initialState) {
    let storeCreator;

    //  client side
    if (typeof(window) !== 'undefined')
    {
        storeCreator = createStore(
            rootReducer,
            initialState,
            compose(
                applyMiddleware(thunk),
                window.devToolsExtension ? window.devToolsExtension() : f => f
            )
        );
    }
    //  server side
    else{
        storeCreator = createStore(
            rootReducer,
            initialState,
            applyMiddleware(thunk)
        );
    }

    return storeCreator;
}