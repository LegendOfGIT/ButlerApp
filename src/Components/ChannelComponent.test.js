import React from 'react';
import ChannelComponent from './ChannelComponent';
import TestRenderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

jest.mock(
    '../Containers/ChannelItemContainer',
    () =>
        { return () => <div>ChannelItemContainer</div>; }
);

const mockFetchChannelDataFn = jest.fn();
jest.mock(
    '../Actions/ChannelActions',
    () => ({
        fetchChannelData: () => ({
            type: 'test'
        })
    })
);

const mockStore = configureStore();
const mockStoreDefault = {
    channel: { }
};

describe('Channel Component', () => {
    it('renders without crashing', () => {
        expect(TestRenderer.create(
            <Provider dispatch={jest.fn()} store={mockStore(mockStoreDefault)}>
                <ChannelComponent/>
            </Provider>
        ).toJSON()).toMatchSnapshot();
    });

    it('shows loading animation, when loading state in store is active', () => {
        const store = mockStore({
            channel: {
                isLoading: true
            }
        });
        expect(TestRenderer.create(
            <Provider dispatch={jest.fn()} store={store}>
                <ChannelComponent/>
            </Provider>
        ).toJSON()).toMatchSnapshot();
    });

    it('shows channel title from store', () => {
        const store = mockStore({
            channel: {
                title: 'Fernsehprogramm: Pro-Sieben'
            }
        });
        expect(TestRenderer.create(
            <Provider dispatch={jest.fn()} store={store}>
                <ChannelComponent/>
            </Provider>
        ).toJSON()).toMatchSnapshot();
    });

    it('shows an error, when error state in store is active', () => {
        const store = mockStore({
            channel: {
                hasError: true
            }
        });
        expect(TestRenderer.create(
            <Provider dispatch={jest.fn()} store={store}>
                <ChannelComponent/>
            </Provider>
        ).toJSON()).toMatchSnapshot();
    });

    it('calls fetchChannelData Action, when component is mount', () => {
        const dispatchStore = mockStore(mockStoreDefault);
        TestRenderer.create(
            <Provider dispatch={jest.fn()} store={dispatchStore}>
                <ChannelComponent />
            </Provider>
        );
        expect(dispatchStore.getActions()).toEqual([{
            type: 'test'
        }]);
    });
});

