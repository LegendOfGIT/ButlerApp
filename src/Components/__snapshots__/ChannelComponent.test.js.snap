// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel Component renders without crashing 1`] = `
<div
  className="Channel"
>
  <h2 />
</div>
`;

exports[`Channel Component shows an error, when error state in store is active 1`] = `
<div
  className="alert alert-danger"
>
  error while loading channel
</div>
`;

exports[`Channel Component shows channel title from store 1`] = `
<div
  className="Channel"
>
  <h2>
    Fernsehprogramm: Pro-Sieben
  </h2>
</div>
`;

exports[`Channel Component shows channel-items from store 1`] = `
<div
  className="Channel"
>
  <h2 />
  <div
    id="AAA-BBB-CCC-DDD"
  >
    ChannelItemContainer
  </div>
  <div
    id="BBB-CCC-DDD-EEE"
  >
    ChannelItemContainer
  </div>
  <div
    id="CCC-DDD-EEE-FFF"
  >
    ChannelItemContainer
  </div>
</div>
`;

exports[`Channel Component shows loading animation, when loading state in store is active 1`] = `
<div
  className="alert alert-info"
>
  Loading channel...
  <div
    className="fa fa-spin fa-spinner"
  >
    Â 
  </div>
</div>
`;
